src/app/Hooks/components/HooksTable.tsx

```diff
@@ -0,0 +1,156 @@
+import TableEmptyState from '@app/common/components/TableEmptyState';
+import HookActionsDropdown from './HookActionsDropdown';
+
+interface IHooksTableProps {
+  hooks: IHook[];
+  openCreateHookModal: () => void;
+  openEditHookModal: (hook: IHook) => void;
+}
+
+const HooksTable: React.FunctionComponent<IHooksTableProps> = ({
+  hooks,
+  openCreateHookModal,
+  openEditHookModal,
+}: IHooksTableProps) => {
+  const filterCategories: FilterCategory<IHook>[] = [
+    {
+      key: 'name',
+      title: 'Name',
+      type: FilterType.search,
+      placeholderText: 'Filter by name...',
+      getItemValue: (item) => {
+        return item.metadata.name;
+      },
+    },
+    {
+      key: 'url',
+      title: 'URL',
+      type: FilterType.search,
+      placeholderText: 'Filter by URL...',
+      getItemValue: (item) => {
+        return item.spec.url;
+      },
+    },
+    {
+      key: 'branch',
+      title: 'Branch',
+      type: FilterType.search,
+      placeholderText: 'Filter by branch...',
+      getItemValue: (item) => {
+        return item.spec.branch;
+      },
+    },
+  ];
+
+  const { filterValues, setFilterValues, filteredItems } = useFilterState(hooks, filterCategories);
+  const getSortValues = (hook: IHook) => [
+    hook.metadata.name,
+    hook.spec.url,
+    hook.spec.branch,
+    '', // plans column,
+    '', // Action column
+  ];
+
+  const { sortBy, onSort, sortedItems } = useSortState(filteredItems, getSortValues);
+  const { currentPageItems, setPageNumber, paginationProps } = usePaginationState(sortedItems, 10);
+  React.useEffect(() => setPageNumber(1), [sortBy, setPageNumber]);
+
+  const columns: ICell[] = [
+    { title: 'Name', transforms: [sortable, wrappable] },
+    { title: 'Git repository URL', transforms: [sortable, wrappable] },
+    { title: 'Branch', transforms: [sortable] },
+    { title: 'Plans', transforms: [cellWidth(20)] },
+    {
+      title: '',
+      transforms: [cellWidth(10)],
+      columnTransforms: [classNames(alignment.textAlignRight)],
+    },
+  ];
+
+  const rows: IRow[] = [];
+
+  currentPageItems.forEach((hook: IHook) => {
+    rows.push({
+      meta: { hook },
+      cells: [
+        hook.metadata.name,
+        hook.spec.url,
+        hook.spec.branch,
+        0,
+        {
+          title: (
+            <HookActionsDropdown hook={hook} openEditHookModal={() => openEditHookModal(hook)} />
+          ),
+        },
+      ],
+    });
+  });
+
+  return (
+    <>
+      <Level>
+        <LevelItem>
+          <Flex>
+            <FlexItem
+              alignSelf={{ default: 'alignSelfFlexStart' }}
+              spacer={{ default: 'spacerNone' }}
+            >
+              <FilterToolbar<IHook>
+                filterCategories={filterCategories}
+                filterValues={filterValues}
+                setFilterValues={setFilterValues}
+              />
+            </FlexItem>
+            <FlexItem>
+              <div>
```

Comment:
Looking at `CreatePlanButton` and `CreateMappingButton`, to align this button with the filter toolbar we may need `className={spacing.mtMd}` on this div. Seems weird though, I'm not sure why it was necessary in those cases. Maybe some props on the `FlexItem` to align it vertically would work too, in which case we should do the same in the other pages. I'm probably being a perfectionist though, the margin class is fine.

Reply:
`spacing.mtMd` fixes it!
I leave addressing it across the board to another PR.